packages/shared/src/hooks/useTopReader.ts

```diff
@@ -0,0 +1,41 @@
+import { useQuery } from '@tanstack/react-query';
+import { generateQueryKey, RequestKey, StaleTime } from '../lib/query';
+import { fetchTopReaderById, fetchTopReaders } from '../lib/topReader';
+import { disabledRefetch } from '../lib/func';
+import type { LoggedUser, PublicProfile } from '../lib/user';
+import type { TopReader } from '../components/badges/TopReaderBadge';
+
+type UseTopReaderProps = {
+  user: LoggedUser | PublicProfile;
+  limit?: number;
+  badgeId?: string;
+};
+
+type UseTopReader = (props: UseTopReaderProps) => {
+  data: TopReader[];
+  isLoading: boolean;
+};
+
+export const useTopReader: UseTopReader = ({ limit = 5, user, badgeId }) => {
+  const { data, isLoading } = useQuery({
```

Comment:
isLoading no longer exists you need to cast is as `isPending`
(isLoading now means `isInitialLoading`